test/Voxel.MiddyNet.ProblemDetails.Tests/ProblemDetailsMiddlewareShould.cs

```diff
@@ -0,0 +1,98 @@
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+using FluentAssertions;
+using Newtonsoft.Json;
+using NSubstitute;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Voxel.MiddyNet.ProblemDetails.Tests
+{
+    public class ProblemDetailsMiddlewareShould
+    {
+        [Fact]
+        public async Task ReturnThrownExceptionsInProblemDetailsFormat()
+        {
+            var function = GivenAFunctionThatThrowsExceptions();
+            var lambdaContext = GivenSomeLambdaContext();
+
+            var response = await function.Handler(new APIGatewayProxyRequest(), lambdaContext);
+
+            response.StatusCode.Should().Be(500);
+            var problemDetails = JsonConvert.DeserializeObject<ProblemDetails>(response.Body);
+            problemDetails.MiddlewareBeforeExceptions.Length.Should().Be(1);
+            problemDetails.MiddlewareAfterExceptions.Length.Should().Be(1);
+            problemDetails.HandlerException.Should().NotBeNull();
+        }
+
+        [Fact]
+        public async Task IncludeStackTraceInProblemDetails()
+        {
+            var function = GivenAFunctionThatThrowsExceptions(true);
```

Comment:
It took me some time to see the difference between the arrange part of this test and the next. 
Maybe the method name could be more verbose?
Ex: `GivenAFunctionThatThrowsExceptionAndTheOptionToIncludeStackTraceIsSetTo(true)`